codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -326,6 +320,56 @@ private int allocateBytesForTree(Http2Stream parent, int connectionWindow) {
     }
 
     /**
+     * Converts the children to an array.
+     */
+    private static Http2Stream[] getChildrenAsArray(Http2Stream parent) {
+        try {
+            final Http2Stream[] children = new Http2Stream[parent.numChildren()];
+            parent.forEachChild(new Http2StreamVisitor() {
+                int ix = 0;
+                @Override
+                public boolean visit(Http2Stream stream) throws Http2Exception {
+                    children[ix++] = stream;
+                    return true;
+                }
+            });
+
+            return children;
+        } catch (Http2Exception e) {
+            // This will never happen.
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * A {@link Http2StreamVisitor} that allocates all of the bytes for each child stream.
+     */
+    private final class AllocationVisitor implements Http2StreamVisitor {
+        int bytesAllocated;
+        int connectionWindow;
+
+        AllocationVisitor(int connectionWindow) {
+            this.connectionWindow = connectionWindow;
+        }
+
+        @Override
+        public boolean visit(Http2Stream child) {
+            FlowState childState = state(child);
+            int bytesForChild = childState.streamableBytes();
+
+            if (bytesForChild > 0 || childState.hasFrame()) {
+                childState.allocate(bytesForChild);
+                bytesAllocated += bytesForChild;
+                connectionWindow -= bytesForChild;
+            }
+            int childBytesAllocated = allocateBytesForTree(child, connectionWindow);
```

Comment:
synthetic generation.  Calling private methods/members from outside scope may result in a synthetic method being generated to access that method/member.